### ISSUE TITLE
Packs install (dowhload) is broken when repo's default branch is not master

### ISSUE THREAD

**Issue by Kami** [StackStorm/st2#3200]
If user specifies a "branch" parameter when installing a pack it doesn't work.

From Slack:

```bash
[14:19] root@ip-10-1-0-25:~/aws/roles/stackstorm/tasks# st2 execution get 589c7a79ecf50606a1314870
id: 589c7a79ecf50606a1314870
status: failed (2s elapsed)
parameters: 
  force: false
  packs:
  - git@github.com:pearsontechnology/stackstorm-bitesize.git
result: 
  exit_code: 1
  result: None
  stderr: "Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 238, in <module>
    obj.run()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 152, in run
    output = action.run(**self._parameters)
  File "/opt/stackstorm/packs/packs/actions/pack_mgmt/download.py", line 75, in run
    verifyssl=verifyssl, ref=pack_version)
  File "/opt/stackstorm/packs/packs/actions/pack_mgmt/download.py", line 139, in _clone_repo
    repo.git.checkout('-b', short_branch, branch)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/git/cmd.py", line 423, in <lambda>
    return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/git/cmd.py", line 868, in _call_process
    return self.execute(call, **_kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/git/cmd.py", line 686, in execute
    raise GitCommandError(command, status, stderr_value, stdout_value)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git checkout -b dev dev
  stderr: 'fatal: A branch named 'dev' already exists.'
"
  stdout: ''
```

We need some unit and integration tests for this as well.

**Comment by Kami** [comment]
Edit: It looks like it might only be related to an edge case when using a non-master default branch - confirming...

**Comment by Kami** [comment]
Confirmed - it's related to the default branch not being master - will work on a fix.

**Comment by Kami** [comment]
Fixed by #3201.
